SolveNQueens

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.4.16 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+contract SolveNQueens {
+
+    TemplateTestContract testContract;  // Create variable for the testContract which needs to be solved.
+    address payable owner;      
+    bool[][]board;              // Chessboard
+    uint8 N;                    // Number of queens  
+
+    constructor(address testAddress) payable {              
+        testContract = TemplateTestContract(testAddress);   // Initialise the testContract variable.
+        owner = msg.sender;                                 // Initialise the owner of the contract to be the creator of the contract.
+        N = 4;
+    }
+
+     // Function to solve the testContract.
+    function solve() public payable{
+        testContract.test(owner);
+    }
+
+    function main(bool[][] memory x) public returns (bool[][] memory b) {
+        (bool a, bool[][] memory solution) = solveNQ(x, N);
+        return(solution);
+
+    }
+
+    function isSafe(bool[][] memory b, uint8 row, uint8 col) public view returns (bool) { 
+        uint8 i = row;
+        uint8 j = col; 
+    
+        // Check this row on left side 
+        for (i = 0; i < col; i++) 
+            if (b[row][i]) 
+                return false; 
+    
+        // Check upper diagonal on left side 
+        for (i;i >= 0; i--) 
+            for (j; j >= 0; j--) 
+                if (b[i][j]) 
+                    return false; 
+        
+        // Check lower diagonal on left side 
+        for (i; i < N; i++) 
+            for (j; j >= 0; j--) 
+                if (b[i][j]) 
+                    return false; 
+    
+        return (true); 
+    } 
+
+    function solveNQ(bool[][] memory b, uint8 col) public returns (bool, bool[][] memory) { 
```

Comment:
Would it perhaps be possible to give a brief explanation on:
0. what the `solveNQ` method does
1. (how that "factors" in in solving the task of the testcontract)
2. What its input parameters are? (Or rename `b` to `board` if that is what it represents)?
3. What it returns?